ui/non_interactive.go

```diff
@@ -40,12 +42,21 @@ func (ui *NonInteractiveUI) PrintTable(table Table) {
 	ui.parent.PrintTable(table)
 }
 
-func (ui *NonInteractiveUI) AskForText(label string) (string, error) {
-	panic("Cannot ask for input in non-interactive UI")
+func (ui *NonInteractiveUI) AskForText(opts TextOpts) (string, error) {
+	if opts.ValidateFunc != nil {
+		isValid, message, err := opts.ValidateFunc(opts.Default)
+		if err != nil || !isValid {
+			return "", fmt.Errorf("Validation Error: %s", message)
```

Comment:
no need to capitalize "Error".

`Validation error: %s`

Reply:
Adopted.